### PR TITLE
add realtime and monotonic clock support for wasi on windows

### LINKED ISSUE #455: support for env_logger not yet implemented?
### Summary

writing a programming using `env_logger` and `log`, but when executing these statement, it panicked with `not yet implemented`

### Additional details

```
$ wasmer run guess_wasi.wasm -- RUST_LOG=DEBUG
Please input your guess.
100
scaned string: ""100\n""
You guessed: 100
too big!
thread 'main' panicked at 'not yet implemented', lib\wasi\src\syscalls\windows.rs:16:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
thread 'main' panicked at 'error: "unhandled trap at 7ff84a9a9129 - code #e06d7363: unknown exception code"', src\bin\wasmer.rs:467:17
stack backtrace:
   0:     0x7ff62a18a5ed - std::sys::windows::backtrace::set_frames
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\sys\windows\backtrace\mod.rs:94
   1:     0x7ff62a18a5ed - std::sys::windows::backtrace::unwind_backtrace
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\sys\windows\backtrace\mod.rs:81
   2:     0x7ff62a18a5ed - std::sys_common::backtrace::_print
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\sys_common\backtrace.rs:70
   3:     0x7ff62a18d9ed - std::sys_common::backtrace::print
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\sys_common\backtrace.rs:58
   4:     0x7ff62a18d9ed - std::panicking::default_hook::{{closure}}
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:200
   5:     0x7ff62a18d670 - std::panicking::default_hook
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:215
   6:     0x7ff62a18e27f - std::panicking::rust_panic_with_hook
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:478
   7:     0x7ff62a18ddb3 - std::panicking::continue_panic_fmt
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:385
   8:     0x7ff62a18dd01 - std::panicking::begin_panic_fmt
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:340
   9:     0x7ff629f5cb7e - <wasmer::Backend as core::str::FromStr>::from_str::hcccf4cab5b2b1dd9
  10:     0x7ff629f60f16 - <wasmer::Backend as core::str::FromStr>::from_str::hcccf4cab5b2b1dd9
  11:     0x7ff629f55bb5 - std::io::read_to_end::h586cee48d6fa7b78
  12:     0x7ff62a18dbf6 - std::rt::lang_start_internal::{{closure}}
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\rt.rs:49
  13:     0x7ff62a18dbf6 - std::panicking::try::do_call<closure,i32>
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:297
  14:     0x7ff62a193021 - panic_unwind::__rust_maybe_catch_panic
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libpanic_unwind\lib.rs:87
  15:     0x7ff62a18e5b1 - std::panicking::try
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panicking.rs:276
  16:     0x7ff62a18e5b1 - std::panic::catch_unwind
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\panic.rs:388
  17:     0x7ff62a18e5b1 - std::rt::lang_start_internal
                               at /rustc/6c2484dc3c532c052f159264e970278d8b77cdc9\/src\libstd\rt.rs:48
  18:     0x7ff629f624b6 - main
  19:     0x7ff62a2024eb - invoke_main
                               at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  20:     0x7ff62a2024eb - __scrt_common_main_seh
                               at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283
  21:     0x7ff84d5e7973 - BaseThreadInitThunk
  22:     0x7ff84dcaa270 - RtlUserThreadStart
thread panicked while panicking. aborting.
Illegal instruction
```

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
resolves #455

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

File: .circleci/config.yml [hunk 1]
```diff
@@ -176,6 +176,10 @@ jobs:
             export LLVM_SYS_70_PREFIX="`pwd`/clang+llvm-7.0.0-x86_64-linux-gnu-ubuntu-16.04/"
             make test-emscripten-clif
             make test-emscripten-llvm
+      - run:
+          name: Debug flag checked
+          command: |
+            cargo check --features "debug"
```
Comment (syrusakbary) [inline .circleci/config.yml]: Good catch!

File: lib/wasi/Cargo.toml [hunk 2]
```diff
@@ -19,6 +19,7 @@ log = "0.4.6"
 byteorder = "1.3.1"
 # hack to get tests to work
 wasmer-singlepass-backend = { path = "../singlepass-backend", version = "0.4.2", optional = true }
+winapi = "0.3"
```
Comment (syrusakbary) [inline lib/wasi/Cargo.toml]: I think we can depend on this only on windows:

```suggestion

[target.'cfg(windows)'.dependencies]
winapi = "0.3"
```

**Pull request merged**